ember-app/app/views/modal.js

```diff
@@ -36,14 +36,22 @@ var ModalView = Ember.View.extend({
     this.$(".fullscreen-overlay, .close").on('click.modal', function(event){
      if(Ember.$(event.target).is("[data-ember-action],[data-toggle]")){return;}
      if(Ember.$(event.target).parents("[data-ember-action],[data-toggle]").length){return;}
-     if(this.modalCloseCriteria()){
-       this.send("modalCloseAction");
-     } else {
-       this.get('controller').send('closeModal');
-     }
+     this.close();
     }.bind(this));
 
     this.$(':input:not(.close):not([type="checkbox"])').first().focus();
+
+    this.$('.modal-close').click(function(){
+      this.close();
+    }.bind(this));
```

Comment:
Not ideal here, I think this should be an action... 

But for now, please add the unsubscribe event in the destroyElement hook